# Problem Set 10
Holly Do

In [1]:
import random as rnd
import math as mth
import statistics as sts
import numpy as np
import scipy as sci
import scipy.linalg as la
import scipy.stats as st
import scipy.special as spp
import matplotlib.pyplot as plt
import sympy as sy
import stemgraphic as stm

import itertools as it
import operator as op

# 1.1

Given: 
* Large sample size of 120 > threshold
* Given a polulation standard deviation $\sigma$

Formula to use : Z distribution

This is a Z test, one sided - right side, test about the mean.

Set up null and alternative hypothesis:

$H_0:\ $  mu = 35

$H_A:\ $ mu > 35

# 1.2

In [30]:
n = 120
mean = 38.4
sigma = 8.6
mu_0 = 35
zstat = (mean - mu_0) / (sigma / mth.sqrt(n))
print(zstat)
pval = 1 - st.norm().cdf(zstat)
pval

4.3308295244594515


np.float64(7.4274329291412755e-06)

P value is very small, which means there is sufficient evidence to reject the null hypothesis and favor the claim that mean number of students 
simultaneously using the Triskellion 4 server is greater than 35.

# 1.3

In [ ]:
alpha = 0.02

At the 2% level of significance, there is sufficient evidence that the mean number of Triskellion 4 
users greater than 35 because pval < 0.02.

# 2

Given: 
* Underlying distribution is normally distributed
* Given a polulation standard deviation $\sigma$

Formula to use : Z distribution

Setup for Hypothesis Test:

$H_0: mu = 40$  

$H_A: mu < 40$

This is a one sided - left side - Z test about the mean.

In [31]:
mullet = np.loadtxt('mullet-update.txt')
sigma = 6 
mu_0 = 40
s_mean = np.mean(mullet) 
n = len(mullet)
zstat = (s_mean - mu_0) / (sigma / mth.sqrt(n))
print(zstat)
pval = st.norm().cdf(zstat)
pval

4.93558882374101


np.float64(0.9999996004534913)

P value is close to 1, which means there is not sufficient evidence to reject the null hypothesis. There is not sufficient evidence to support the vendor claim that the mean update time for this rev is less than 40 minutes.

# 3

Given: 
* Small sample size of 5
* Not given a polulation standard deviation $\sigma$

Formula to use : T distribution

This is a T test about a mean, left sided test. 

Setup for Hypothesis Test:

$H_0: mu = 75$  

$H_A: mu < 75$

In [27]:
n = 5
rate = np.array([35,45,60,75,78])
mu = 75
s_sig = np.std(rate, ddof=1)
xbar = np.mean(rate)
print(xbar)
tstat = (xbar - mu)/(s_sig/mth.sqrt(n))
print(tstat)
dof = n-1
dst = st.t(dof).cdf(tstat)
pval

58.6
-1.9677796210229768


np.float64(0.12047557286818411)

P value for this test is 0.12, if alpha is set to be less than 0.12 then there is not significant evidence to support the claim. If alpha is set to be larger than 0.12 then there is significant evidence to support the claim.

# 4.1

Given: 
* Large sample size of 1000 > threshold
* Proportion

Formula to use : Z distribution


### Notes
What test? Looking for proportion.  
Sample size > threshold
So, Z-test.

What do we Know:
$p_0 = .1$

Setup for Hypothesis Test:

$\hat{p}hat = 115/1000$

$H_0: p = .1$  
$H_A: p > .1$

In [37]:
# Z-test for proportion
ndefects = 115
n = 1000 # sample size
phat = ndefects/n
p0 = 0.1
sig2 = (p0*(1-p0))/n
sig = mth.sqrt(sig2)
zstat = (phat-p0)/sig
pval = 1 - st.norm().cdf(zstat)
pval

np.float64(0.056923149003329065)

At 5% level of significance, pval is slightly too large to reject null hypothesis.
So, Mr. Coyote cannot dispute the 10% failure rate that ACME claims at this level of significance.

# 4.2

In [ ]:
alpha = 0.2

At 20% level of significance, pval is too large to reject null hypothesis.
So, Mr. Coyote cannot dispute the 10% failure rate that ACME claims at this level of significance.

# 5

Given: 
* Large sample size of 350 and 400 > threshold
* Proportion

Formula to use : Z distribution

Problem class = diff of proportions.  

So, Z-test. (sample sizes large enough)

Z-Test difference of proportions setup:

$H_0 : $  phat_H - phat_D = 0

$H_A : $  phat_H - phat_D < 0 

Z distribution 1 sided test of difference in proportions

In [38]:
#
# "_H" for Houston
# "_D" for Dallas
#
ndefects_H = 10
n_H = 350
phat_H = ndefects_H/n_H

ndefects_D = 16
n_D = 400
phat_D = ndefects_D/n_D

## pooled sample variance
p0 = (ndefects_H + ndefects_D)/(n_H + n_D)
sig0 = p0*(1-p0)
sig = mth.sqrt(sig0*((1./n_H) + (1./n_D)))

zstat = (phat_H - phat_D)/sig
print(zstat)
pval = st.norm().cdf(zstat)
pval

-0.8535532875814258


np.float64(0.19667627511559216)

At the 5% level of significance, pval 0.1967 is greater than 0.05,  so the Houston office cannot claim a better process than Dallas office.

# 5.2

In [ ]:
alpha = 0.1

At the 10% level of significance, pval 0.1967 is greater than 0.1,  so the Houston office cannot claim a better process than Dallas office.

# 6

Given: 
* Large sample size of 84 and 200 > threshold 50
* Proportion

Formula to use : Z distribution

Problem class = diff of proportions.  

Z-Test difference of proportions setup:

$H_0 : $  phat_SLO - phat_S = 0

$H_A : $  phat_SLO - phat_S < 0 

Z test 1 sided - left side - test of difference in proportions

In [32]:
#
# "_SLO" for San Lois Obispo
# "_S" for Sacramento
#
nvote_S = 60
n_S = 84
phat_S = nvote_S/n_S

nvote_SLO = 118
n_SLO = 200
phat_SLO = nvote_SLO/n_SLO

## pooled sample variance
p0 = (nvote_S + nvote_SLO)/(n_S + n_SLO)
sig0 = p0*(1-p0)
sig = mth.sqrt(sig0*((1./n_S) + (1./n_SLO)))

zstat = (phat_SLO - phat_S)/sig
pval = st.norm().cdf(zstat)
pval

np.float64(0.02405545535223496)

Pval is 0.024. If pval > alpha he cannot claim level of support in San Luis Obispo is significantly less than his support in Sacramento. If pval < alpha he can claim level of support in San Luis Obispo is significantly less than his support in Sacramento.

# 7

Given: 
* Large sample size of 57 > threshold 50
* Not given a trusted population standard deviation 

Formula to use : T distribution

# 7.1 
Design an appropriate statistical test to determine if the mean difference between Frank and Drac is 0 or not. Using the monster.txt data, give the P-value for your test.


$H_0 : $  mu = 0

$H_A : $  mu != 0 

T distribution test about the mean - 2 sided test

In [40]:
monster = np.loadtxt('monster.txt').astype(int)
n = len(monster)
xbar = np.mean(monster)
sigma = np.std(monster,ddof=1)
mu = 0
tstat = (xbar - mu)/(sigma/mth.sqrt(n))
dof = n-1
dst = 1-st.t(dof).cdf(abs(tstat))
pval = 2 * dst
pval

np.float64(0.8076028068906813)

Pval is 0.8, if alpha is set to be less than 0.8 then there is not significant evidence to support the claim. If alpha is set to be larger than 0.8 then there is significant evidence to support the claim.

# 7.2
Suppose Sara is of the opinion that Frank is “slightly” better than Drac. This means that the average of the Frank-Drac data should be negative. Give the P-value for this new test. Note: RANKING IS ORDINAL so Frank-Drac is negative!

Ha changed, from 2 sided to 1 sided - left sided - T test about the mean.


$H_0 : $  mu = 0

$H_A : $  mu < 0 


In [39]:
monster = np.loadtxt('monster.txt').astype(int)
n = len(monster)
xbar = np.mean(monster)
sigma = np.std(monster,ddof=1)
mu = 0
tstat = (xbar - mu)/(sigma/mth.sqrt(n))
print(tstat)
dof = n-1
dst = st.t(dof).cdf(tstat)
pval = dst
pval

-0.2446744936015633


np.float64(0.4038014034453407)

Pval is 0.4, if alpha is set to be less than 0.4 then there is not significant evidence to support the claim. If alpha is set to be larger than 0.4 then there is significant evidence to support the claim.

# 8

Given: 
* Small sample size of 6 < threshold
* Not given a population standard deviation 

Formula to use : T distribution

In [4]:
anthony = np.array([85,92,97,65,75,96])
eric = np.array([81,79,76,84,83,77])

# 8.1

Given: 
* Actual scores of the two friends are independent and normally distributed
* Not given a population standard deviation 

Formula to use : T distribution

This is a one sided - right side - T test of difference between two means.

Assume the variances in the student’s grades are equal.

$H_0 : $  mu_anthony - mu_eric = 0

$H_A : $  mu_anthony - mu_eric > 0 


## Pooled Sample Variance (or Std Deviation)
When conducting a 2-sample (or paired sample) t-test, 1 possible assumption is that the variances of the paired
samples are unknown, but equal. To get a better estimate for the (common) variance, the best method is to combine
the 2 samples into a common variance estimate called the *pooled sample variance*, denoted as $s_p^2$.
The formula for the pooled sample variance for 2 samples $X,Y$
>$
    s_p^2 = \dfrac{{\displaystyle \sum_{i=1}^n} (x_i - \bar{X})^2 + 
    {\displaystyle \sum_{i=1}^m }(y_i - \bar{Y})^2}{n+m-2}
$  

where $n$ is the size of the $X$ sample, and $m$ is the size of the $Y$ sample.

The components of the numerator of $s_p^2$ are the sum-of-squared differences used in the calculation of
the variance (or sample variance). Consequently, the formula for $s_p^2$ can be written as
>$
    s_p^2 = \dfrac{(n)\sigma^2(X) + (m)\sigma^2(Y)}{n+m-2}
$

In [16]:
anthony_mean = np.mean(anthony)
eric_mean = np.mean(eric)
anthony_var = np.var(anthony, ddof = 1)
eric_var = np.var(eric, ddof = 1)
na = 6
ne = 6

## pooled sample variance
spool2 = ((na*anthony_var) + (ne*eric_var)) / (na + ne -2)
spool = mth.sqrt(spool2)
sig = spool*mth.sqrt((1./na) + (1./ne))

tstat = (anthony_mean - eric_mean - 0)/sig
print(tstat)
df = 6 + 6 - 2 #10
pval = 1 - st.t(df).cdf(zstat)
pval

0.849534582538231


np.float64(0.18698577208106237)

Pval is 0.19, if alpha is set to be less than 0.19 then there is not significant evidence to support Anthony’s claim. If alpha is set to be larger than 0.19 then there is significant evidence to support Anthony’s claim.

# 8.2

Assume the variances in the student’s grades are not equal.

$H_0 : $  mu_anthony - mu_eric = 0

$H_A : $  mu_anthony - mu_eric > 0 

Formula to use : T distribution
This is a one sided - right side - T test of difference between two means.

## 2-Sample t-test With Unequal Variance (test difference of means)
The t-statistic for a difference of means assuming the variances are unequal (and unknown) is:
>$
\dfrac{\bar{X} - \bar{Y} - D}{\sqrt{\frac{s_X^2}{n} + \frac{s_Y^2}{m}}}
$

where:
- the 2 samples are $X,Y$
- the sample means are $\bar{X},\bar{Y}$
- the null hypothesis is that $\mu_X - \mu_Y = D$.  
    NOTE: $D = 0$ is a common case
- the size of the sample from $X.Y$ is $n,m$ respectively
- $s_X^2,s_Y^2$ are the sample variances

In this unequal-variance case, the degrees of freedom for the t-test is given by the Satterthwaite formula
>$
   \nu = \dfrac{\left(\dfrac{s_X^2}{n} + \dfrac{s_Y^2}{m}\right)^2}
   {\dfrac{s_X^4}{n^2\,(n-1)} + \dfrac{s_Y^4}{m^2\,(m-1)}}
$

In [36]:
anthony_mean = np.mean(anthony)
print(anthony_mean)
eric_mean = np.mean(eric)
print(eric_mean)
anthony_var = np.var(anthony, ddof = 1)
eric_var = np.var(eric, ddof = 1)
na = 6
ne = 6

## pooled sample variance
sig = mth.sqrt((anthony_var/na) + (eric_var/ne))

tstat = (anthony_mean - eric_mean)/sig

dftop = ((anthony_var/na) + (eric_var/ne))**2 
dfbottom = ((anthony_var**2) / ((na**2)*(na-1))) + ((eric_var**2) / ((ne**2)*(ne-1))) #parentheses
df = dftop/dfbottom
print(df)

pval = 1-st.t(df).cdf(tstat)
pval

85.0
80.0
5.636224259732453


np.float64(0.19508584405029739)

Pval is 0.195, if alpha is set to be less than 0.195 then there is not significant evidence to support Anthony’s claim. If alpha is set to be larger than 0.195 then there is significant evidence to support Anthony’s claim.